### ISSUE TITLE
Floating point FBO in SDL_gpu

### ISSUE THREAD

**Issue by NTxC** [grimfang4/sdl-gpu#235]
I was wondering whether SDL_gpu supports floating point framebuffer objects.

I'm working on a remake of an older game in C/C++ and I used SDL_gpu to be able to utilize shaders. However, when I draw using the shader of my choice (ntsc-adaptive from libretro), everything seems to have a pink tint to it (left - shader off, right - shader on):

![off-on](https://user-images.githubusercontent.com/25325789/166950649-eb13123e-0974-4920-8edf-8ecb5fe042d2.jpg)

I learned that this might be a side effect of not utilizing a floating point FBO. Does SDL_gpu support this in some way? When calling GPU_CreateImage I can pick the format, but whatever SDL_gpu format I use, it all seems to be of type GL_UNSIGNED_BYTE under the hood (and I'm assuming I need GL_FLOAT).

**Comment by grimfang4** [comment]
That is correct.  SDL_gpu assumes GL_UNSIGNED_BYTE.  Floating point framebuffers would probably be a good feature to expose, as lacking it limits things like HDR shaders.  I can't tell if that's your actual issue here, but it is an issue.

**Comment by NTxC** [comment]
I see, thanks. SDL_gpu doesn't seem to enjoy more than 4 bytes per pixel under the hood too, so quickly adding in GL_RGBA32F/GL_FLOAT doesn't seem like it's going to work.

Great library, by the way. Really fun to use.

**Comment by NTxC** [comment]
I can confirm the issue was indeed the lack of floating point framebuffer object support in SDL_gpu. I implemented that functionality (GL_RGBA32F support) myself and it's all working now.

![IM2](https://github.com/grimfang4/sdl-gpu/assets/25325789/2105ca8b-0885-4ab6-946c-fdf7e4e5b08a)

I might create a pull request later.
